### PR TITLE
Make size_of tests measure stylo and reduce the size of stylo PropertyDeclarations by 16 bytes

### PR DESCRIPTION
**Pull request by bholley**
Right now they don't, which means that we have four properties making PropertyDeclaration 16 bytes bigger than it should be.

I'm not sure if there's a better way to get these tests to run against stylo than to hoist them into the properties file, but I couldn't figure it out. This seems good enough.

---
This change is [Reviewable](https://reviewable.io/reviews/servo/servo/15953)

### CODE REVIEW

**Comment by bholley** [comment]
I guess the other question is whether non-stylo test runs actually run tests from components/style or whether they're strictly limited to tests/unit/style. We certainly shouldn't disable these tests for normal servo runs.

CC @Wafflespeanut

**Comment by bholley** [comment]
Also, the _reason_ those properties are so large is that we have a bunch of extra fields on URL specified values for stylo.

File: components/style/properties/properties.mako.rs [hunk 1]
```diff
@@ -2401,3 +2400,51 @@ pub fn specified_value_sizes() -> Vec<(&'static str, usize, bool)> {
...
+
+#[test]
+fn size_of_property_declaration() {
+    use std::mem::size_of;
+
+    let old = 48;
+    let new = size_of::<PropertyDeclaration>();
+    if new < old {
+        panic!("Your changes have decreased the stack size of PropertyDeclaration enum from {} to {}. \
+                Good work! Please update the size in tests/unit/style/size_of.rs.",
```
Comment (jdm) [inline components/style/properties/properties.mako.rs]: This message references the old file.

File: components/style/properties/properties.mako.rs [hunk 2]
```diff
@@ -2401,3 +2400,51 @@ pub fn specified_value_sizes() -> Vec<(&'static str, usize, bool)> {
...
+                old, new)
+    } else if new > old {
+        panic!("Your changes have increased the stack size of PropertyDeclaration enum from {} to {}. \
+                These enum is present in large quantities in the style, and increasing the size \
+                may negatively affect style system performance. Please consider using `boxed=\"True\"` in \
+                the longhand If you feel that the increase is necessary, update to the new size in \
+                tests/unit/style/size_of.rs.",
```
Comment (jdm) [inline components/style/properties/properties.mako.rs]: This message references the old file.

**Review by jdm — CHANGES_REQUESTED** [review]

**Comment by Manishearth** [comment]
> I guess the other question is whether non-stylo test runs actually run tests from components/style or whether they're strictly limited to tests/unit/style

It doesn't. There aren't supposed to be tests in components/, the only reason style is an exception is that bindgen generates layout tests inline and we have to run those for stylo. So you probably want to export a function that can be called from tests/unit and components/style

**Comment by Wafflespeanut** [comment]
We should consider having another `size_of` module in `stylo_tests` and tweak the the features to enable both `testing` and `gecko` for the `style` crate. Then, we can test this using `./mach test-stylo`

**Comment by SimonSapin** [comment]
@bholley This PR adds a `#[test]` function in the style crate, but I think that test will never be run in our current CI. To run it you’d need to change something in `python/servo` so that either `./mach test-unit` or `./mach test-stylo` also runs `cargo test -p style`.

However that involves recompiling the entire style crate (with `--test` passed to rustc) which take some time. To avoid that the test can be in a separate crate, either similar to `tests/unit/style` (as a Cargo "package" with its own `Cargo.toml` file) or what Cargo calls an [integration test](http://doc.crates.io/manifest.html#integration-tests) (which is a separate crate from the point of view of rustc, but not a separate package as far as Cargo is concerned).

**Comment by Wafflespeanut** [comment]
... or, simply make a copy of `size_of` (or move it) to stylo tests, so that it can be tested there.

**Comment by bholley** [comment]
Fixed up. r? @Manishearth

**Pull request merged**